client.go

```diff
@@ -486,8 +486,10 @@ func (c *Client) send(r *Request) (*amqp.Delivery, error) {
 		r.Timeout = c.timeout
 	}
 
-	// start the timeout counting now.
-	timeoutChan := r.startTimeout()
+	var timeoutChan <-chan time.Time
```

Comment:
Is this required? We won’t end up in the select case if we don’t want reply. I can’t see how this affects the deadline on the request? The timeout value is unchanged, this will just don’t stop the timer which doesn’t matter?

Reply:
> We won’t end up in the select case if we don’t want reply.

Are you sure? When we don't want a reply a `nil` response is sent back on the `responses` channel after the publishing has been sent to rabbit.

> I can’t see how this affects the deadline on the request? The timeout value is unchanged, this will just don’t stop the timer which doesn’t matter?

When running `r.startTimeout()` the `Expiration` is set on the publishing at the same time as the timeout counter is started. But since we don't want to have a timeout, we just never run it.

Reply:
Oh, sorry! You're right!

I would rather do this (to skip the `var -> if` thing which I don't like) in `request.go`:

```
func (r *Request) startTimeout() <-chan time.Time {
	if r.Reply {
		r.Publishing.Expiration = fmt.Sprintf("%d", r.Timeout.Nanoseconds()/1e6)
	}

	return time.After(r.Timeout)
}
```

Reply:
That's better yes. :+1: